File: Novels/packages/6-mobile-android/StoryReader/app/src/main/java/com/storysphere/storyreader/model/Wallet.kt
package com.storysphere.storyreader.model

data class Wallet(
    val userId: String,
    val balance: Int = 0, // Points/Coins balance
    val currency: Currency = Currency.POINTS,
    val lastUpdatedAt: Long = System.currentTimeMillis()
)

enum class Currency {
    POINTS, // Virtual currency (1 point = 1/100 CNY)
    COINS   // Alternative currency name
}

data class Transaction(
    val id: String,
    val userId: String,
    val type: TransactionType,
    val amount: Int,
    val currency: Currency = Currency.POINTS,
    val description: String? = null,
    val relatedId: String? = null, // Story ID, Chapter ID, etc.
    val createdAt: Long = System.currentTimeMillis()
)

enum class TransactionType {
    TOP_UP,        // Top-up wallet
    PURCHASE,      // Purchase chapter
    REWARD,        // Reward from missions
    REFUND,        // Refund
    TRANSFER       // Transfer to/from
}

File: Novels/packages/5-mobile-ios/StoryReader/Model/Wallet.swift
import Foundation

// Wallet model - Virtual Currency
struct Wallet: Codable {
    let balance: Int // Points balance (1 point = 0.01 USD)
    let currency: String
    let updatedAt: Date?
    
    enum CodingKeys: String, CodingKey {
        case balance, currency
        case updatedAt = "updated_at"
    }
}

// Transaction model
struct Transaction: Codable, Identifiable {
    let id: String
    let type: TransactionType
    let amount: Int
    let description: String?
    let createdAt: Date
    
    enum TransactionType: String, Codable {
        case topUp = "top_up"
        case purchase = "purchase"
        case refund = "refund"
    }
    
    enum CodingKeys: String, CodingKey {
        case id, type, amount, description
        case createdAt = "created_at"
    }
}

